From 41251182e9f228835b226eb581ea1d8a7d27b613 Mon Sep 17 00:00:00 2001
From: Neil Horman <nhorman@tuxdriver.com>
Date: Tue, 2 Apr 2013 15:07:59 -0400
Subject: [PATCH 50/53] sctp: Add new spinfo state values to enumeration

A while back the kernel added two enumeration values to sctp_spinfo_state.
Unfortunately, those were never added to the user space header file, and as a
result we get inconsistent behavior when reading spinfo state.  This patch adds
the missing fields in, and corrects the behavior.

Signed-off-by: Neil Horman <nhorman@tuxdriver.com>
CC: William Reich <reich@ulticom.com>
CC: Daniel Borkmann <dborkman@redhat.com>
CC: Vlad Yasevich <vyasevich@gmail.com>
Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
---
 src/include/netinet/sctp.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/include/netinet/sctp.h b/src/include/netinet/sctp.h
index 14fa604..51f5bfb 100644
--- a/src/include/netinet/sctp.h
+++ b/src/include/netinet/sctp.h
@@ -676,8 +676,10 @@ struct sctp_paddrinfo {
  */
 enum sctp_spinfo_state {
 	SCTP_INACTIVE,
+	SCTP_PF,
 	SCTP_ACTIVE,
-	SCTP_UNCONFIRMED
+	SCTP_UNCONFIRMED,
+	SCTP_UNKNOWN = 0xffff
 };
 
 /*
-- 
1.7.11.7

